### PR TITLE
Moths.

### PR DESCRIPTION
**Pull request by LankLTE**
## About the PR

Adds moths as a playable roundstart race. At the moment a basic implementation, hence the draft. Go yell at me to fix it, change it, or whatever else you have. 
Moths are simple: 
They can only eat clothing and cloth. 
They can fly in zero gravity, granting them better acceleration and maneuverability
Take more damage from heat, less from cold. 

I hope to make them eat clothes and disallow their wings from working in space because.. that's how physics works. 

Guess I'm rewriting species yaml now. Sloths' list: 
- [x]  use parent composition for body parts
- [x] remove iconcomponents
- [x] use parent composition for the human + NPC variants
- [x] Parent composition for organs

do the above for other species too

- [x] Add proper flight, carpet eating
- [ ] (optional) have some spreadsheet outlining how moths are different so then we can add how other species are different and shit
(it's simple anyways i'm not doing the bottom one)

Once this is implemented more markings can be added. What clothes they can eat was pretty generously applied, I'll change it if it becomes a problem.

**Media**

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

(I'll add more pictures when this works better) 

image

https://github.com/space-wizards/space-station-14/assets/135308300/8f7c0cf1-a928-41b7-af0f-82ebf5fe3167

**Changelog**

:cl: Lank
- add: Added Moth People as a playable race. These friends can move around in zero gravity easily and love the cold, but are significantly weaker against high temperatures and won't be caught eating anything not made of cloth.

### CODE REVIEW

**Comment by Aisu9** [comment]
i had the idea to let moths toggle a personal no gravity mode while inside environment with gravity, only because it would be fun to see a moth flying in the hallways of the station and crashing headfirst on a wall or something

**Comment by crazybrain23** [comment]
> Can move around in zero gravity (including space).

https://youtu.be/TJ8O8Ovi2CY

**Comment by metalgearsloth** [comment]
From current freezes:

> Roundstart / player species, such as lizards / humans. A) These need design docs B) There's existing species in 13 we may want to port instead C) the maintenance burden is enormous and not feasible for a lot of species.

C) in particular needs fixing first.

**Comment by Emisse** [comment]
this has been given the greenlight given species get cleaned up and duplication is murdered across different playable species

**Comment by UbaserB** [comment]
moffs my beloved

**Comment by deltanedas** [comment]
tokyo drift

**Comment by lzk228** [comment]
NO PLEASE
just let's wait for vounds rework and then merge new races

**Comment by Myakot** [comment]
Consdering recent merge of #18566 + that video of a butterfly in space I'm getting eager to see players straight up smashing their heads in against walls and fuel tanks, prompting explosions. Glorious.

**Comment by LankLTE** [comment]
> Consdering recent merge of #18566 + that video of a butterfly in space I'm getting eager to see players straight up smashing their heads in against walls and fuel tanks, prompting explosions. Glorious.

Zero gravity speed is still capped, so sadly you won't be seeing this anytime soon.

**Comment by LankLTE** [comment]
I (think) we're all good to go. Moth enjoyers rejoice, or whatever.

File: Content.Server/Nutrition/EntitySystems/FoodSystem.cs
Comment (metalgearsloth) [inline Content.Server/Nutrition/EntitySystems/FoodSystem.cs:113]: Don't hardcode checks in here, use events and another system as mentioned above.

Also do not use Console.Write, use the logger.

File: Content.Server/Nutrition/Components/FoodComponent.cs
Comment (metalgearsloth) [inline Content.Server/Nutrition/Components/FoodComponent.cs:82]: Make this a separate FoodWhitelist component or do some metabolism or whatever.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
Use BaseX rather than XBase.

Rename all of the Human stuff as it's not human-specific.

Some of the composition is skrunkly as you're using MobHuman as a parent but then that also drags along BaseMobHuman too which shouldn't be happening as now everything has to inherit humans and is likely going to cause issues. Split the player stuff into its own prototype and use that.

File: Resources/Prototypes/Entities/Mobs/Species/moth.yml
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Mobs/Species/moth.yml:61]: Alphabetical please.

File: Resources/Prototypes/Entities/Mobs/Species/reptilian.yml
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Mobs/Species/reptilian.yml:16]: ?

File: Resources/Prototypes/tags.yml
Comment (metalgearsloth) [inline Resources/Prototypes/tags.yml:889]: Alphabetical please like above (ignore flareguy's addition).

File: Resources/Textures/Mobs/Customization/moth_parts.rsi/meta.json
Comment (metalgearsloth) [inline Resources/Textures/Mobs/Customization/moth_parts.rsi/meta.json:4]: This needs to point to a specific commit.

**Comment by CaptainSqrBeard** [comment]
Isn't moths were something banned?
https://github.com/space-wizards/space-station-14/pull/12055#issuecomment-1284413916

File: Resources/Prototypes/Entities/Mobs/Species/human.yml
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Mobs/Species/human.yml:15]: Are you sure this is even needed with https://github.com/space-wizards/space-station-14/pull/18291

**Comment by Equivocateur** [comment]
> Isn't moths were something banned? [#12055 (comment)](https://github.com/space-wizards/space-station-14/pull/12055#issuecomment-1284413916)

notreal

**Review by lzk228 — CHANGES_REQUESTED** [review]
dataset with moth names?

File: Resources/Prototypes/Entities/Mobs/Species/human.yml
Comment (LankLTE) [reply Resources/Prototypes/Entities/Mobs/Species/human.yml:15]: That isn’t merged yet, but you also explicitly told me earlier than the NPCs needed it still even with that PR.

**Comment by Vordenburg** [comment]
> > Use BaseX rather than XBase.
> 
> What does this mean

`BaseHead` not `HeadBase` I presume.

**Comment by lzk228** [comment]
> > dataset with moth names?
> 
> I made a set of moth names downstream. I could PR it here if you like.

maybe send it to author?

**Comment by Vordenburg** [comment]
There you go: #18618

File: Resources/Prototypes/Entities/Mobs/Species/human.yml
Comment (metalgearsloth) [reply Resources/Prototypes/Entities/Mobs/Species/human.yml:15]: Maybe I was thinking of it being on MobHuman but that's probably fine then.

**Comment by metalgearsloth** [comment]
lol

**Review by metalgearsloth — APPROVED** [review]

**Comment by LankLTE** [comment]
MOTHS!!! THE MOTHS ARE REAL!!!!

**Comment by BasedUser** [comment]
any% coding speedrun

**Comment by PuroSlavKing** [comment]
awesome work man ❤️ but I would change the shape of the eyes

**Pull request merged**
